File: conformance/bundles/17-unexpected-pass/QuietSteps.java
package dev.varar.conformance.bundle17;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;

/**
 * Java sibling of {@code quiet.steps.ts} (bundle {@code 17-unexpected-pass}).
 *
 * <p>The example carries an {@code error} fence, so it asserts a failure. This stimulus throws
 * nothing, so the fence inverts into an UnexpectedPassError — the kind no bundle exercised before
 * this one.
 */
public final class QuietSteps implements StepDefinitions<QuietSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.state(Ctx::new);

        s.stimulus("I do nothing at all", (Ctx ctx) -> ctx);
    }
}

File: conformance/bundles/17-unexpected-pass/quiet.steps.rs
//! Rust sibling of `quiet.steps.ts` (bundle `17-unexpected-pass`).
//!
//! The example carries an `error` fence, so it asserts a failure. This stimulus
//! panics nothing, so the fence inverts into an UnexpectedPassError — the kind
//! no bundle exercised before this one.

use varar::Steps;

pub fn register(s: &mut Steps<()>) {
    s.stimulus("I do nothing at all", |ctx: ()| Ok(ctx));
}

pub fn state() {}
